### PR TITLE
Support underscores in identifiers

### PR DESCRIPTION
**Pull request by marcauberer**
- Fixes the issue of #46 for the C++ compiler
- Updated grammar accordingly
- Grammar fixes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -250,17 +250,18 @@ COMP_STMT             --> KEY COMP_OP VALUE
 CONTAINS_STMT         --> KEY not? contains KEY COMP_OP VALUE
 KEY                   --> IDENTIFIER INDEX? (.IDENTIFIER INDEX?)*
 INDEX                 --> [NUMBER]
-IDENTIFIER            --> LETTER+
+IDENTIFIER            --> LETTER (LETTER* DIGIT* UNDERSCORE*)* | UNDERSCORE (LETTER* DIGIT* UNDERSCORE*)*
```
Comment (Splines) [inline README.md:253]: `_` would be a valid identifier then. Do we allow this?

---

We could simplify this rule to:

```
IDENTIFIER --> (LETTER|UNDERSCORE) (LETTER* DIGIT* UNDERSCORE*)*
````

**Review by Splines — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (marcauberer) [reply README.md:253]: Technically, I see no problem with allowing this, so I would allow it.
Thanks for the simplification of the grammar statement

**Comment by marcauberer** [comment]
@Splines everything should be fixed now. Can you approve?

**Review by Splines — APPROVED** [review]
👌

**Pull request merged**